frontend/src/Frontend/UI/Dialogs/WatchRequest.hs

```diff
@@ -59,59 +68,86 @@ inputRequestKey
+        Pact.PactResult (Right a) -> Pact.renderCompactText a
+
+  close <- modalHeader $ text "Check TX Status"
+
+  modalMain $ do
     dialogSectionHeading mempty "Notice"
     divClass "group" $ do
       text "If you have a request key from a previously submitted transaction, you can use this dialog to wait for and display the results."
+
     dialogSectionHeading mempty "Required Information"
-    info <- divClass "group" $ do
-      mChain <- fmap value $ mkLabeledClsInput False "Chain ID" $
-        uiChainSelection (headMay <$> nodes) (pure Nothing)
-      eKey <- mkLabeledInput False "Request Key" uiInputElement def
-      let parseRequestKey t
-            | Right v <- Pact.fromText' t = Right v
-            | otherwise = Left "Invalid hash"
-      pure $ liftA2 (,) (current mChain) (fmap parseRequestKey . current $ value eKey)
-    void $ runWithReplace blank $ ffor err $ \e -> do
-      dialogSectionHeading mempty "Error"
-      divClass "group" $ text e
-    pure info
-  (cancel, next) <- modalFooter $ do
-    cancel' <- cancelButton def "Cancel"
-    next' <- confirmButton def "Next"
-    pure (cancel', next')
-  let f (Just c, Right k) = Right (c, k)
-      f (_, Left e) = Left e
-      f (Nothing, _) = Left "You must select a chain"
-      (err, ok) = fanEither $ attachWith (const . f) choice next
-      nextScreen = attachWith (watchRequestKey onCloseExternal) (current nodes) ok
-  pure ((mempty, close <> cancel), nextScreen)
-
-watchRequestKey
-  :: (Monoid mConf, MonadWidget t m)
-  => Event t () -- ^ Modal was externally closed
-  -> [NodeInfo] -- ^ Nodes to try
-  -> (ChainId, Pact.RequestKey) -- ^ User selected chain and request key
-  -> Workflow t m (mConf, Event t ())
-watchRequestKey onCloseExternal nodes (chain, reqKey) = Workflow $ mdo -- can't use 'rec' due to GHC bug
-  close <- modalHeader $ text "Transaction Status"
-  pb <- getPostBuild
-  let clientEnvs = mkClientEnvs nodes chain
-  (listenStatus, message, _) <- listenToRequestKey clientEnvs $ leftmost
-    [ Just reqKey <$ (pb <> reloadKey)
-    , Nothing <$ (closeModal <> onCloseExternal) -- Cancel any inflight request when the modal is closed
-    ]
-  reloadKey <- elClass "div" "modal__main transaction_details" $ do
-    mkLabeledInput True "Transaction Hash" uiInputElement $ def
-      & initialAttributes .~ "disabled" =: "disabled"
-      & inputElementConfig_initialValue .~ Pact.requestKeyToB16Text reqKey
-    reload <- transactionStatusSection (pure Status_Done) listenStatus
-    transactionResultSection message
-    pure reload
+    dRqKey <- (fmap . fmap) hush $ divClass "group" $ fmap snd $ uiRequestKeyInput False
+
+    rec
+      let lockRefresh = (||) <$> fmap isNothing dRqKey <*> blockSpam
+
+      onRequest <- divClass "group check-tx-status__button-wrapper" $
+        confirmButton (def & uiButtonCfg_disabled .~ lockRefresh) checkTxBtnLbl
+
+      onTimeoutFinished <- delay 5 onRequest
+
+      blockSpam <- toggle False $ leftmost
```

Comment:
This would be more accurate as
```haskell
blockSpam <- holdDyn False $ leftmost
  [ True <$ onRequest
  , False <$ onTimeoutFinished
  ]
```
It seems equivalent if we never have more than one request in transit, but I don't think we want to add implicit assumptions on that.